rpc/src/client/transport/router.rs

```diff
@@ -17,36 +19,54 @@ pub struct SubscriptionRouter {
     // their result channels. Used for publishing events relating to a specific
     // query.
     subscriptions: HashMap<String, HashMap<String, SubscriptionTx>>,
```

Comment:
We should probably use type aliases or even new types here, eg.

```suggestion
    subscriptions: HashMap<SubscriptionQuery, HashMap<SubscriptionId, SubscriptionTx>>,
```

Reply:
Makes sense. I do have an issue open for reintroducing query parsing into the codebase (#859) that'll provide the `Query` type.